server/src/main/java/org/elasticsearch/index/mapper/AbstractPointGeometryFieldMapper.java

```diff
@@ -120,8 +120,12 @@ public void parse(XContentParser parser, CheckedConsumer<T, IOException> consume
                     consumer.accept(validate(point));
                 } else {
                     while (token != XContentParser.Token.END_ARRAY) {
-                        parseAndConsumeFromObject(parser, point, consumer, onMalformed);
-                        point = pointSupplier.get();
+                        if (parser.currentToken() == XContentParser.Token.VALUE_NULL) {
+                            consumer.accept(nullValue);
```

Comment:
I feel like this should be wrapped into `if (nullValue != null) { ... } ` otherwise it causes NPE inside consumer that it interprets as malformed data.

Reply:
:+1: I see how the other code had it. I added a test for this.